doc/smart-switch/pmon/smartswitch-pmon.md

```diff
@@ -0,0 +1,947 @@
+* The switch will fetch the reboot-cause history from each of the DPUs as needed when the "show reboot-cause history" CLI is issued on the switch.
+
+#### 3.4.2 Reboot Cause CLIs on the DPUs      <font>**`Executed on the DPU`**</font>
+* The "show reboot-cause" shows the most recent reboot-cause of th
+* The "show reboot-cause history" shows the reboot-cause history
+```
+root@sonic:~#show reboot-cause
+
+Name                    Cause                       Time                                User    Comment
+
+2023_10_02_17_20_46     reboot                      Sun 02 Oct 2023 05:20:46 PM UTC     admin   User issued 'reboot'
+
+root@sonic:~#show reboot-cause history
+
+Name                    Cause                       Time                                User    Comment
+
+2023_10_02_17_20_46     reboot                      Sun 02 Oct 2023 05:20:46 PM UTC     admin   User issued 'reboot'
+2023_10_02_18_10_00     reboot                      Sun 02 Oct 2023 06:10:00 PM UTC     admin   User issued 'reboot'
+```
+#### 3.4.3 Reboot Cause CLIs on the Switch      <font>**`Executed on the switch`**</font>
+* The "show reboot-cause" CLI on the switch shows the most recent rebooted device, time and the cause. The could be the NPU or any DPU
+* The "show reboot-cause history" CLI on the switch shows the history of the Switch and all DPUs
+* The "show reboot-cause history module-name" CLI on the switch shows the history of the specified module
+
+"show reboot-cause history"      <font>**`Executed on the switch`**</font>
+```
+root@sonic:~#show reboot-cause
+
+Device          Name                    Cause                       Time                                User    Comment
+
+switch          2023_10_20_18_52_28     Watchdog:1 expired;         Wed 20 Oct 2023 06:52:28 PM UTC     N/A     N/A
+
+root@sonic:~#show reboot-cause history
+
+Device          Name                    Cause                       Time                                User    Comment
+
+switch          2023_10_20_18_52_28     Watchdog:1 expired;         Wed 20 Oct 2023 06:52:28 PM UTC     N/A     N/A
+switch          2023_10_05_18_23_46     reboot                      Wed 05 Oct 2023 06:23:46 PM UTC     user    N/A
+DPU0            2023_10_04_18_23_46     Power Loss                  Tue 04 Oct 2023 06:23:46 PM UTC     N/A     N/A
+DPU3            2023_10_03_18_23_46     Watchdog: stage 1 expired;  Mon 03 Oct 2023 06:23:46 PM UTC     N/A     N/A
+DPU3            2023_10_02_18_23_46     Host Power-cycle            Sun 02 Oct 2023 06:23:46 PM UTC     N/A     Host lost DPU
+DPU3            2023_10_02_17_23_46     Host Reset DPU              Sun 02 Oct 2023 05:23:46 PM UTC     N/A     N/A
+DPU2            2023_10_02_17_20_46     reboot                      Sun 02 Oct 2023 05:20:46 PM UTC     admin   User issued 'reboot'
+
+"show reboot-cause history <module-name>"
```

Comment:
Is <module-name> option only for DPU modules? Is it possible to specify 'switch' as the module-name?